### ISSUE TITLE
Password problems with RTL after Upgrade on a running Umbrel system

### ISSUE THREAD

**Issue by Gravity-87** [Ride-The-Lightning/RTL#968]
**Describe the bug**
After the actual Umbrel upgrade to v0.4.15 an user-set password for login to RTL is lost/dropped. There's only access to RTL with the installation default password, which is not shown, if you upgrade a running Umbrel/RTL installation. The login password also seems not to be changeable on Umbrel shell level. I suspect, there goes something wrong with the $APP_PASSWORD string?! But I don't know exactly....

**To Reproduce**
Login failure appears always as long you try to get RTL-access with your old password and don't know about a reset to a default password during the upgrade procedure... 

**Screenshots**
/

**Your environment**
* Version of `RTL`: 0.12.1-beta
* Version of `lnd`: 0.14.2-beta 
* Version of `btcd`, `bitcoind`, or other backend: v22.0.0
* Browser & browser version: Firefox v 97.0.1
* Operating system (`uname -a` on *Nix): Linux umbrel 5.10.63-v8+ #1459 SMP PREEMPT Wed Oct 6 16:42:49 BST 2021 aarch64 GNU/Linux
* any other relevant environment details: Umbrel v0.4.15

**Additional context**
A simple-dimple solution is, to uninstall RTL from the Umbrel system and reinstall it from the Umbrel App store. During this procedure you get at least the default password displayed, with which access to the RTL installtion is possible.

It seems so, that in the actual setup of Umbrel/RTL there is no possibility to set the RTL-password to an individual phrase, neither via RTL-app nor on the grounds of the Umbrel shell level.

My preferred solution would be: Changing option for the login password within the RTL app. Especially, the way of changing the user password during a kind of 'headless install procedure' (cause of shutting down Umbrel/RTL within this procedure) seems not to be very user-friendly. ;-)

Reference to:
https://community.getumbrel.com/t/ride-the-lightning-password-changed/6568

Thanx a lot!

**Comment by saubyk** [comment]
Hi @Gravity-87 
There is no need to delete and re-install RTL after the upgrade. You can just use the password, available in the App Store to login. With this update, RTL user will not be able to set a custom password and will have to use the password set by the Umbrel store.

**Comment by tjw74** [comment]
Why? 

How is this a good thing to use a password provided by a 3rd party? I don't understand what we're doing unless you are saying there is a bug regarding the password and the only way to use RTL for now is to use the default password? 

App uninstalled.

**Comment by saubyk** [comment]
Hi @tjw74 
There is a nuance here, which has resulted in this kind of setup. Umbrel is orchestrating user passwords and passing the default password as an environment variable for RTL to authenticate the user with. Ideally, you should petition Umbrel to give you the flexibility to clear or change the application passwords, so that application security can work independently.

Additionally, you can also turn on 2FA in RTL, to ensure that you are not using the 3rd party supplied password only to authenticate yourselves.

Overall, I would say that this not an ideal setup from a UX as well as security standpoint and should definitely be improved. But the changes cannot be driven by us only, would require collaboration with Umbrel as well.

Whether you want to use the app or not, over this, is entirely your decision. :-)

Thanks

**Comment by tjw74** [comment]
Fair enough, thank you for clarifying. Umbrel is messing up here.

**Comment by Gravity-87** [comment]
I just wrote it in the Umbrel issue section, because it seems to be better located there:
"OK, the default password is displayed in the app store. It's not really necessary to un-/reinstall the app. Checked!
But what happens, if you connect via Tor Browser to your Umbrel? I just experienced, that the app store isn't reachable then at all. So, you don't get access to the default password there. Seems to be a kind of suboptimal solution furthermore."
